cmd/cli/version/version.go

```diff
@@ -63,7 +65,10 @@ func NewCmd() *cobra.Command {
 
 func runVersion(cmd *cobra.Command, oV *VersionOptions) error {
 	ctx := cmd.Context()
-
+	INSTALLATIONID := os.Getenv("BACALHAU_USER_INSTALLATIONID")
+	if INSTALLATIONID != "" && config.GetInstallationUserID()==""{
+		config.SetIntallationID(INSTALLATIONID)
+	}
```

Comment:
You shouldn't need to do this explicitly – the config system should already be picking it up for you. Can you try removing this and doing this test: if you run Bacalhau without an established config file, does the installation ID set by this environment variable automatically get persisted to that config file?

You can test this in a fresh config location using e.g.
```bash
export BACALHAU_DIR=$(mktemp -d)
export BACALHAU_USER_INSTALLATIONID="aaaaa"
bacalhau version
cat "$BACALHAU_DIR/config.yaml"
```

Reply:
Setting an environment variable will not cause changes to the config file if that's what's being suggested here?

Reply:
But if the environment variable was set before starting bacalhau `config.GetInstallationUserID` will return that value, so we don't need to make the `os.Getenv("BACALHAU_USER_INSTALLATIONID")` call here. Its handled by the config

Reply:
> export BACALHAU_DIR=$(mktemp -d)
> export BACALHAU_USER_INSTALLATIONID="aaaaa"
> bacalhau version
> cat "$BACALHAU_DIR/config.yaml"
The main issue with this is it doesn't show up because the binary that's installed in install.sh isn't build from scratch. so there is no way of knowing it except for merging and releasing the code or modifying the code in install.sh to build from scratch

Reply:
Ah, I was under the impression that the config that is written out on first init of a repo will represent the current state of config set by environment variables. But yeah, this isn't the case. So we do need to set the installation ID persistently somehow.